keywords/LiteServAndroid.py

```diff
@@ -1,83 +1,41 @@
 import subprocess
 
-import requests
-
 from keywords.LiteServBase import LiteServBase
-from keywords.constants import LATEST_BUILDS
-from keywords.constants import BINARY_DIR
-from keywords.constants import REGISTERED_CLIENT_DBS
 from keywords.exceptions import LiteServError
-from keywords.utils import version_and_build
 from keywords.utils import log_info
 
 
 class LiteServAndroid(LiteServBase):
 
-    def download(self):
-        """
-        1. Check to see if .apk is downloaded already. If so, return
-        2. Download the LiteServ .apk from latest builds to 'deps/binaries'
-        """
-
-        version, build = version_and_build(self.version_build)
-
-        if version == "1.2.1":
-            package_name = "couchbase-lite-android-liteserv-SQLite-{}-debug.apk".format(self.version_build)
-        else:
-            if self.storage_engine == "SQLite":
-                package_name = "couchbase-lite-android-liteserv-SQLite-{}-debug.apk".format(self.version_build)
-            else:
-                package_name = "couchbase-lite-android-liteserv-SQLCipher-ForestDB-Encryption-{}-debug.apk".format(self.version_build)
-
-        expected_binary_path = "{}/{}".format(BINARY_DIR, package_name)
-        if os.path.isfile(expected_binary_path):
-            log_info("Package is already downloaded. Skipping.")
-            return
-
-        # Package not downloaded, proceed to download from latest builds
-        if version == "1.2.1":
-            url = "{}/couchbase-lite-android/release/{}/{}/{}".format(LATEST_BUILDS, version, self.version_build, package_name)
-        else:
-            url = "{}/couchbase-lite-android/{}/{}/{}".format(LATEST_BUILDS, version, self.version_build, package_name)
-
-        log_info("Downloading {} -> {}/{}".format(url, BINARY_DIR, package_name))
-        resp = requests.get(url)
-        resp.raise_for_status()
-        with open("{}/{}".format(BINARY_DIR, package_name), "wb") as f:
-            f.write(resp.content)
-
-    def install(self):
+    def install_apk(self, apk_path, apk_id):
         """Install the apk to running Android device or emulator"""
-
-        if self.storage_engine == "SQLite":
-            apk_name = "couchbase-lite-android-liteserv-SQLite-{}-debug.apk".format(self.version_build)
-        else:
-            apk_name = "couchbase-lite-android-liteserv-SQLCipher-ForestDB-Encryption-{}-debug.apk".format(self.version_build)
-
-        apk_path = "{}/{}".format(BINARY_DIR, apk_name)
         log_info("Installing: {}".format(apk_path))
 
-        output = subprocess.check_output(["adb", "install", apk_path])
-        if "INSTALL_FAILED_ALREADY_EXISTS" in output or "INSTALL_FAILED_UPDATE_INCOMPATIBLE" in output:
-            raise LiteServError("Error. APK already exists.")
+        process = subprocess.Popen(["adb", "install", apk_path], stdout=subprocess.PIPE)
+        output, _ = process.communicate()
+        process.poll()
+        if "INSTALL_FAILED_ALREADY_EXISTS" in output or \
```

Comment:
>>> import subprocess
>>> apk_path
'/Users/raghu.sarangapani/Downloads/LiteServ/Android/LiteServ.apk'
>>> process = subprocess.Popen(["adb", "install", apk_path], stdout=subprocess.PIPE)
>>> output, _ = process.communicate()
Success
>>> output
''
>>> process = subprocess.Popen(["adb", "install", apk_path], stdout=subprocess.PIPE)
>>> output, _ = process.communicate()
Failed to install /Users/raghu.sarangapani/Downloads/LiteServ/Android/LiteServ.apk: Failure [INSTALL_FAILED_ALREADY_EXISTS: Attempt to re-install com.couchbase.liteserv without first uninstalling.]
>>> output
''

The output from adb is not captured in "output" variable.